src/components/SearchBar/SearchBar.hook.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, type ChangeEvent, type KeyboardEvent } from 'react';
+
+import { DEBOUNCE_DELAY } from './SearchBar.constants';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const useSearchBar = (props: SearchBarProps) => {
+  const {
+    value: controlledValue,
+    defaultValue,
+    onSearch,
+    debounceDelay = DEBOUNCE_DELAY,
+    onChange,
+  } = props;
+
+  const isControlled = controlledValue !== undefined;
+  const [internalValue, setInternalValue] = useState(defaultValue || '');
+
+  // Sync internal state with controlled value
+  useEffect(() => {
+    if (isControlled) {
+      setInternalValue(controlledValue);
+    }
+  }, [controlledValue, isControlled]);
+
+  // Debounce logic
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      // Only trigger search if not controlled or if explicit requirement
+      // Usually debounce is for auto-search on type.
+      // Requirement says "debounce... onSearch seguro".
+      // If we want auto-search on type:
+      if (onSearch && internalValue) {
+        // onSearch(internalValue); // Uncomment if auto-search is desired
+      }
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [internalValue, debounceDelay, onSearch]);
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Handlers (handleInputChange) are recreated on every render. When this hook is used inside a component, passing new handler references can trigger unnecessary re-renders of child components. Memoize handlers with useCallback and properly declare dependencies.

**Sugerencia:**
```typescript
Wrap the handler in useCallback: const handleInputChange = useCallback((e) => { ... }, [isControlled, onChange]);
```